NPE in NewCUProposal.createChange(NewCUProposal.java:277)

```
[Error - 11:14:07 AM] Request textDocument/codeAction failed.
  Message: Internal error.
  Code: -32603 
java.util.concurrent.CompletionException: java.lang.NullPointerException
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:645)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:479)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NullPointerException
	at org.eclipse.jdt.ls.core.internal.corrections.proposals.NewCUProposal.createChange(NewCUProposal.java:277)
	at org.eclipse.jdt.core.manipulation.ChangeCorrectionProposalCore.getChange(ChangeCorrectionProposalCore.java:150)
	at org.eclipse.jdt.ls.core.internal.handlers.CodeActionHandler.getCodeActionFromProposal(CodeActionHandler.java:228)
	at org.eclipse.jdt.ls.core.internal.handlers.CodeActionHandler.getCodeActionCommands(CodeActionHandler.java:194)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$14(JDTLanguageServer.java:646)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:75)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
	... 6 more
```

using invisible project with java.project.referencedLibraries with org.apache.commons:commons-math3:3.6.1
trying to call completion after `new org.apache.commons.math3.fraction.`

not sure if related but there is no completion provided after `new org.apache.commons.math3.util.` even if it provides completion for `ContinuedFraction` when starting to type `new Continued` which is in package `org.apache.commons.math3.util`

tried in VS Code Java

--- a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/NewCUProposal.java
+++ b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/NewCUProposal.java
@@ -27,6 +27,7 @@ import org.eclipse.jdt.core.Flags;
 import org.eclipse.jdt.core.ICompilationUnit;
 import org.eclipse.jdt.core.IJavaElement;
 import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
 import org.eclipse.jdt.core.IType;
 import org.eclipse.jdt.core.dom.AST;
 import org.eclipse.jdt.core.dom.ASTNode;
@@ -268,11 +269,10 @@ public class NewCUProposal extends ChangeCorrectionProposal {
 			TextEdit edit = constructEnclosingTypeEdit(parentCU);
 			cuChange.setEdit(edit);
 			return cuChange;
-		} else if (fTypeContainer instanceof IPackageFragment) {
+		} else if (fTypeContainer instanceof IPackageFragment && ((IPackageFragment) fTypeContainer).getKind() == IPackageFragmentRoot.K_SOURCE) {
 			String name = ASTNodes.getSimpleNameIdentifier(fNode);
 			ICompilationUnit parentCU = ((IPackageFragment) fTypeContainer).getCompilationUnit(getCompilationUnitName(name));
 			targetType = parentCU.getType(name);
-
 			CompositeChange change = new CompositeChange(fName);
 			change.add(new CreateFileChange(targetType.getResource().getRawLocation(), "", ""));
 			change.add(constructNewCUChange(parentCU));

--- a/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandlerTest.java
+++ b/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandlerTest.java
@@ -292,6 +292,30 @@ public class CodeActionHandlerTest extends AbstractCompilationUnitBasedTest {
 		assertTrue("refactor actions should be ahead of source actions",  baseKinds.lastIndexOf(CodeActionKind.Refactor) < baseKinds.indexOf(CodeActionKind.Source));
 	}
 
+	@Test
+	public void testCodeAction_NPEInNewCUProposal() throws Exception {
+		ICompilationUnit unit = getWorkingCopy(
+		//@formatter:off
+				"src/org/sample/Foo.java",
+				"package org.sample;\n"+
+				"public class Foo {\n"+
+				"	public static void main(String[] args) {\n"+
+				"		new javax.activity\n"+
+				"	}\n"+
+				"}\n");
+		//@formatter:off
+		CodeActionParams params = new CodeActionParams();
+		params.setTextDocument(new TextDocumentIdentifier(JDTUtils.toURI(unit)));
+		final Range range = CodeActionUtil.getRange(unit, "javax.activity");
+		params.setRange(range);
+		CodeActionContext context = new CodeActionContext(
+			Arrays.asList(getDiagnostic(Integer.toString(IProblem.UndefinedType), range))
+		);
+		params.setContext(context);
+		List<Either<Command, CodeAction>> codeActions = getCodeActions(params);
+		Assert.assertNotNull(codeActions);
+	}
+
 	private static String getBaseKind(String codeActionKind) {
 		if (codeActionKind.contains(".")) {
 			return codeActionKind.substring(0, codeActionKind.indexOf('.'));

